<launch>

    <arg name="color" default="blue" />
    <param name="color" value="$(arg color)"/>

  <arg name="robot" default="localhost" doc="allows the user to specify the hostname of the turtlebot"/>
  <arg name="scene" default='nuwall' doc="draw the actual obstacle locations and walls"/>

      <group if="$(eval arg('robot') == 'localhost')">
        <machine name="turtlebot" address="michaelangelo" env-loader="$(find nuturtle_control)/loader.sh" user="msr"/>
        <node machine="turtlebot" pkg="rosserial_python" type="serial_node.py" name="serial_node">
            <param name="port" value="/dev/ttyACM0"/>
            <param name="baud" value="115200"/>
        </node>
     </group>
     
      <group if="$(eval arg('robot') != 'localhost')">
        <machine name="turtlebot" address="michaelangelo" env-loader="/home/msr/install/share/nuturtle_control/loader.sh" user="msr"/>
        <node machine="turtlebot" pkg="hls_lfcd_lds_driver" type="hlds_laser_publisher" name="hlds_laser_publisher"></node>
        <node machine="turtlebot" pkg="rosserial_python" type="serial_node.py" name="serial_node">
            <param name="port" value="/dev/ttyACM0"/>
            <param name="baud" value="115200"/>
        </node>
    
    <node pkg="nusim" type="nusim" name="nusim" output="screen" ns="red">
		    <remap from="/red/sensor_data" to="/sensor_data"/>
		    <remap from="/red/wheel_cmd" to="/wheel_cmd"/>
		    <remap from="/red/obstacles" to="/obstacles"/>
            <remap from="/$(arg color)/joint_states" to="/joint_states"/> 
		    <remap from="/red/Wall" to="/Wall"/>
            <param name="scene" value="$(arg scene)"/>
    </node>

     <node machine="turtlebot" pkg="nuturtle_control" type="turtle_interface" name="turtle_interface" ns="$(arg color)" output="screen">
    
    <remap from="/$(arg color)/cmd_vel" to="/cmd_vel"/>   
    <remap from="/$(arg color)/sensor_data" to="/sensor_data"/>
    <remap from="/$(arg color)/wheel_cmd" to="/wheel_cmd"/>  
    <remap from="/$(arg color)/joint_states" to="/joint_states"/> 
    
    </node>
     
    <node machine="turtlebot" pkg="nuturtle_control" type="odometry" name="odometry" ns="$(arg color)" output="screen">
    
    <remap from="/$(arg color)/odom" to="/odom"/>
    <remap from="/$(arg color)/joint_states" to="/joint_states"/> 
    </node>
    </group>

</launch>
